components/style/sharing/mod.rs

```diff
@@ -228,11 +431,6 @@ impl<E: TElement> StyleSharingCandidateCache<E> {
             return StyleSharingResult::CannotShare
         }
 
-        if element.get_id().is_some() {
```

Comment:
Wait what? Where did this come from? This seems like something out of Boris' patch.